client/transport.go

```diff
@@ -0,0 +1,31 @@
+package client
+
+import (
+	"net/http"
+
+	"github.com/fullstorydev/hauser/config"
+)
+
+// FullStory is an HTTP transport which wraps the underlying transport and
+// sets the Authorization header
+type FullStory struct {
+	Key               string
+	AdditionalHeaders []config.Header
+
+	// Transport is the underlying HTTP transport.
+	// if nil, http.DefaultTransport is used.
+	Transport http.RoundTripper
+}
+
+func (t *FullStory) RoundTrip(req *http.Request) (*http.Response, error) {
+	rt := t.Transport
+	if rt == nil {
+		rt = http.DefaultTransport
+	}
+	newReq := *req
```

Comment:
There's not much point in copying the request this way. The only thing you are touching is in headers, and `newReq` and `req` still share the same headers instance since you aren't doing a deep copy. And, FWIW, I don't think you need to do a deep copy. I think you could just add the headers directly to `req`.